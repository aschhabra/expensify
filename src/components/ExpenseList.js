import React from "react";
import {connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import getVisibleExpense from "./../selectors/expenses";

export const ExpenseList= (props) => (
    <div>
      <h1> Expense List</h1>
      { props.expenses.length===0 ?
        (
          <p> there is no Expense in the list</p>
        ):
        (
          props.expenses.map((expense)=> {
            return <ExpenseListItem key={expense.id} {...expense} />;
          })
        )
      }
    </div>

);

const ConnectedExpenseList = connect((state)=>{
  return {
    expenses: getVisibleExpense(state.expenses,state.filters)
  };
})(ExpenseList);


export default ConnectedExpenseList;
