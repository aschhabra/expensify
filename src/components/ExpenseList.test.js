import React from "react";
import {shallow} from "enzyme";
import {ExpenseList} from "./ExpenseList";
import expenses from "../fixtures/expenses"

test( "test should render ExpenseList with expenses",()=>{
  const wrapper= shallow(<ExpenseList expenses={expenses}/>);


  expect(wrapper).toMatchSnapshot();
});
test( "test should render ExpenseList without expenses",()=>{
  const wrapper= shallow(<ExpenseList expenses={[]}/>);


  expect(wrapper).toMatchSnapshot();
});
