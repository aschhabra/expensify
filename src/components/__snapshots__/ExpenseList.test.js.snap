// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should render ExpenseList with expenses 1`] = `
<div>
  <h1>
     Expense List
  </h1>
  <ExpenseListItem
    amount={195}
    createdAt={-432000000}
    description="Gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={4695}
    createdAt={345600000}
    description="Rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={4595}
    createdAt={-345600000}
    description="Test "
    id="3"
    key="3"
    note=""
  />
</div>
`;

exports[`test should render ExpenseList without expenses 1`] = `
<div>
  <h1>
     Expense List
  </h1>
  <p>
     there is no Expense in the list
  </p>
</div>
`;
